File: backend/debug_email.py
import os
import sys
from dotenv import load_dotenv

# Add the current directory to sys.path to import local modules
sys.path.append(os.path.dirname(os.path.abspath(__file__)))

from utils.email_utils import send_otp_email

load_dotenv()

def test_production_email():
    if len(sys.argv) < 2:
        print("Usage: python backend/debug_email.py <recipient_email>")
        sys.exit(1)
        
    recipient = sys.argv[1]
    otp = "123456"

    print("--- Brevo API Diagnostic ---")
    print(f"Testing production OTP delivery to: {recipient}")
    print(f"Using Sender: {os.getenv('BREVO_SENDER', 'learnitfast6@gmail.com')}")
    print(f"API Key present: {'YES' if os.getenv('BREVO_API_KEY') else 'NO'}")
    print("-----------------------------")
    
    success = send_otp_email(recipient, otp)
    
    if success:
        print("SUCCESS: The email was sent successfully via Brevo API!")
    else:
        print("FAILED: Check your .env credentials and BREVO_API_KEY.")

if __name__ == "__main__":
    test_production_email()

File: backend/src/scripts/debugEmail.js
/**
 * scripts/debugEmail.js
 * Test production OTP delivery to a specified recipient email.
 * Usage: node src/scripts/debugEmail.js <recipient_email>
 */

import dotenv from 'dotenv';
import { sendOtpEmail } from '../utils/email.js';

dotenv.config();

async function main() {
  const recipient = process.argv[2];
  if (!recipient) {
    console.log('Usage: node src/scripts/debugEmail.js <recipient_email>');
    process.exit(1);
  }

  console.log('--- Brevo API Diagnostic ---');
  console.log(`Testing production OTP delivery to: ${recipient}`);
  console.log(`Using Sender: ${process.env.BREVO_SENDER || 'learnitfast6@gmail.com'}`);
  console.log(`API Key present: ${process.env.BREVO_API_KEY ? 'YES' : 'NO'}`);
  console.log('-----------------------------');

  const success = await sendOtpEmail(recipient, '123456');

  if (success) {
    console.log('SUCCESS: The email was sent successfully via Brevo API!');
  } else {
    console.log('FAILED: Check your .env credentials and BREVO_API_KEY.');
  }
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
